src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -42,6 +45,13 @@ use crate::utils::{
     ColIndexMapping, ColIndexMappingRewriteExt, Condition, GroupBy, IndexSet, Substitute,
 };
 
+pub struct SeparatedAggInfo {
+    pub approx_percentile_agg_calls: Vec<PlanAggCall>,
+    pub non_approx_percentile_agg_calls: Vec<PlanAggCall>,
+    pub approx_percentile_col_mapping: ColIndexMapping,
+    pub non_approx_percentile_col_mapping: ColIndexMapping,
+}
```

Comment:
Feeling a bit strange but no strong opinion.

```rs
pub struct AggInfo {
    pub calls: Vec<PlanAggCall>,
    pub col_mapping: ColIndexMapping,
}

// return (AggInfo, AggInfo) in `separate_normal_and_special_agg`
```

Or

```rs
pub struct SeparatedAggInfo {
    normal: AggInfo,
    approx: AggInfo,
}
```

may be better?

Reply:
Agree this is better.